[Lockdown Mode] Disable SVG fonts in lockdown mode

litherum opened the pull request:
#### 1d6daa42a5fab36faadc14f8d38d558644d7c88e
<pre>
[Lockdown Mode] Disable SVG fonts in lockdown mode
<a href="https://bugs.webkit.org/show_bug.cgi?id=242981">https://bugs.webkit.org/show_bug.cgi?id=242981</a>
&lt;rdar://problem/92179697&gt;

Reviewed by Geoffrey Garen and Brent Fulgham.

Webfonts are supposed to be disabled in lockdown mode, including SVG fonts.

(Most of the TestWebKitAPI changes in this patch are just messing with headers
because we use unified builds.)

Test: CaptivePortal.Fonts

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/xcshareddata/xcschemes/TestWebKitAPI.xcscheme:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalModeFonts.mm: Added.
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Helvetica_light.svg: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBPersistence.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SVGFont.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPrinting.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewConfiguration.mm:
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:

Canonical link: <a href="https://commits.webkit.org/252712@main">https://commits.webkit.org/252712@main</a>
</pre>

geoffreygaren commented on Source/WebCore/css/CSSFontFace.cpp:
```diff
@@ -69,16 +69,17 @@ void CSSFontFace::appendSources(CSSFontFace& fontFace, CSSValueList& srcList, Sc
         foundSVGFont = item.isSVGFontFaceSrc() || item.svgFontFaceElement();
         fontFaceElement = item.svgFontFaceElement();
+        const auto* settings = context ? &context->settingsValues() : nullptr;
+        bool allowDownloading = settings && settings->downloadableBinaryFontsEnabled;
```
I would have written this as:

bool allowDownloading = false;
if (context && context->settingsValues()->downloadableBinaryFontsEnabled)
    allowDownloading = true;

...because I always find ?: confusing, and this formulation avoids a local pointer variable of indeterminate lifetime.

litherum replied:
Wouldn't it be better to say
```
bool allowDownloading = context && context->settingsValues()->downloadableBinaryFontsEnabled;
```
?

geoffreygaren replied:
I like that too!

geoffreygaren reviewed (approved):
r=me

brentfulgham commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -3542,12 +3549,12 @@
 				F422A3E8235ACA9B00CF00CA /* ClipboardTests.mm */,
 				CDF92236216D186400647AA7 /* CloseWebViewAfterEnterFullscreen.mm */,
 				CDF0B789216D484300421ECC /* CloseWebViewDuringEnterFullscreen.mm */,
-				F4DBE5DC278906A300642BC0 /* CloseWhileCommittingLoad.mm */,
```
I wonder why Xcode removed this file from the project?

litherum replied:
🤔

brentfulgham reviewed (approved):
LGTM.

litherum committed:
[Lockdown Mode] Disable SVG fonts in lockdown mode
https://bugs.webkit.org/show_bug.cgi?id=242981
<rdar://problem/92179697>

Reviewed by Geoffrey Garen and Brent Fulgham.

Webfonts are supposed to be disabled in lockdown mode, including SVG fonts.

(Most of the TestWebKitAPI changes in this patch are just messing with headers
because we use unified builds.)

Test: CaptivePortal.Fonts

* Source/WebCore/css/CSSFontFace.cpp:
(WebCore::CSSFontFace::appendSources):
* Tools/TestWebKitAPI/SourcesCocoa.txt:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/xcshareddata/xcschemes/TestWebKitAPI.xcscheme:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/CaptivePortalModeFonts.mm: Added.
(TestWebKitAPI::TEST):
* Tools/TestWebKitAPI/Tests/WebKitCocoa/Helvetica_light.svg: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/IndexedDBPersistence.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/SVGFont.html: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKPrinting.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/WKWebViewConfiguration.mm:
* Tools/TestWebKitAPI/cocoa/HTTPServer.mm:

Canonical link: https://commits.webkit.org/252712@main
